src/App.tsx

```diff
@@ -1,26 +1,314 @@
+
+    const trimmedTitle = newTodoTitle.trim();
+
+    if (!trimmedTitle) {
+      setErrorMessage(Error.Empty);
+      setTimeout(() => setErrorMessage(Error.Default), 3000);
+
+      return;
+    }
+
+    const loadingTodo: Todo = {
+      userId: USER_ID,
+      title: trimmedTitle,
+      completed: false,
+      id: 0,
+    };
+
+    setTempTodo(loadingTodo);
+    addPost(loadingTodo);
+  }
+
+  const deletePost = useCallback((todoId: number) => {
+    setLoadingTodoId(todoId);
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setErrorMessage(Error.Delete);
+        setTimeout(() => setErrorMessage(Error.Default), 3000);
+      })
+      .finally(() => {
+        setLoadingTodoId(null);
+        inputRef.current?.focus();
+      });
+  }, []);
+
+  const deleteCompleted = useCallback(() => {
+    const completedIds = todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    completedIds.forEach(id => deletePost(id));
+  }, [todos, deletePost]);
+
+  function togglCompleted(todoId: number) {
+    const currentTodo = todos.find(todo => todo.id === todoId);
+
+    if (!currentTodo) {
+      return;
+    }
+
+    const newCompletionState = !currentTodo.completed;
+
+    setLoadingTodoId(todoId);
+
+    updateTodo(todoId, newCompletionState, currentTodo.title)
+      .then(() => {
+        setTodos(prevTodos =>
+          prevTodos.map(todo =>
+            todo.id === todoId
+              ? { ...todo, completed: newCompletionState }
+              : todo,
+          ),
+        );
+      })
+      .catch(() => {
+        setErrorMessage(Error.Update);
+        setTimeout(() => setErrorMessage(Error.Default), 3000);
+      })
+      .finally(() => {
+        setLoadingTodoId(null);
+      });
+  }
+
+  async function completeAll() {
+    const allCompleted = todos.every(todo => todo.completed);
+    const newCompletionState = !allCompleted;
+
+    const todosToUpdate = todos.filter(
+      todo => todo.completed !== newCompletionState,
+    );
+
+    const failedIds: number[] = [];
+
+    for (const todo of todosToUpdate) {
+      try {
+        await updateTodo(todo.id, newCompletionState, todo.title);
+      } catch {
+        failedIds.push(todo.id);
+      }
+    }
+
+    setTodos(currentTodos =>
+      currentTodos.map(todo =>
+        failedIds.includes(todo.id)
+          ? todo
+          : { ...todo, completed: newCompletionState },
+      ),
+    );
+
+    if (failedIds.length > 0) {
+      setErrorMessage(Error.Update);
+    }
+  }
+
+  const updateTitle = useCallback(
+    (todoId: number) => {
+      const trimmedTitle = editingTitle.trim();
+      const currentTodo = todos.find(todo => todo.id === todoId);
+
+      if (currentTodo && trimmedTitle === currentTodo.title) {
+        setEditingTodoId(null);
+
+        return;
+      }
+
+      if (trimmedTitle === '') {
```

Comment:
the same syntax, but more readable
```suggestion
      if (!trimmedTitle.length) {
```